### PR TITLE
kdotool: init at 0.2.1

### LINKED ISSUE #303268: Package request: kdotool
**Project description**
A xdotool clone for KDE Wayland

**Metadata**

* homepage URL: https://github.com/jinliu/kdotool
* source URL: https://github.com/jinliu/kdotool
* license: apache2
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by kotatsuyaki**
## Description of changes

Verified that it works on Plasma `6.0.4`.

Closes #303268

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by SuperSandro2000 — COMMENTED** [review]
Does the tool even have a dependency on qt or kde libraries? If not, we could just place it outside of kdePackages, as we don't really want to add new things to plasma5.

**Comment by K900** [comment]
It should probably go into by-name, yeah

File: pkgs/desktops/plasma-5/3rdparty/kdotool/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+let
+  pname = "kdotool";
+  version = "0.2.1";
+in rustPlatform.buildRustPackage {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "jinliu";
+    repo = pname;
```
[the remark this answers is not in the document]
Comment (kotatsuyaki) [reply pkgs/desktops/plasma-5/3rdparty/kdotool/default.nix]: Replaced `pname` with a hardcoded string.

**Comment by kotatsuyaki** [comment]
Thanks, I have moved it to by-name.

File: pkgs/by-name/kd/kdotool/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{ lib,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  dbus
+}:
+
+let
+  pname = "kdotool";
```
Comment (K900) [inline pkgs/by-name/kd/kdotool/package.nix:9]: Can we move this into the package definition? I'm not sure if the Rust builders do `finalAttrs` but preferably that.

**Pull request merged**